File: appverifier/src/test/java/com/android/mdl/appreader/certutil/CertificateMaterial.java
package com.android.mdl.appreader.certutil;

import java.math.BigInteger;
import java.util.Collections;
import java.util.Date;
import java.util.List;
import java.util.Optional;

public interface CertificateMaterial {	
	BigInteger serialNumber();
	Date startDate();
	Date endDate();
	int keyUsage();
	
	default List<String> extendedKeyUsage() {
	    return Collections.emptyList();
	}
	
    default boolean extendedKeyUsageCritical() {
        return true;
    }

    boolean isCa();
    
    Optional<Integer> pathLengthConstraint();
    
    default boolean includeKeyIds() {
        return true;
    }
    
    default List<String> criticalPolicyOids() {
        return Collections.emptyList();
    }
    
}	

File: appverifier/src/main/java/com/android/mdl/appreader/readercertgen/CertificateMaterial.kt
package com.android.mdl.appreader.readercertgen

import java.math.BigInteger
import java.util.Date
import java.util.Optional

data class CertificateMaterial(
    val serialNumber: BigInteger,
    val startDate: Date,
    val endDate: Date,
    val keyUsage: Int,
    val extendedKeyUsage: Optional<String>,
    val pathLengthConstraint: Int,
) {
    companion object {
        const val PATHLENGTH_NOT_A_CA = -1
    }
}
